### ISSUE TITLE
Final Project Grade

### ISSUE THREAD

**Issue by grader-cogs108** [COGS108/Group134_WI26#10]
# Final Project Grade

## Score (out of 18 pts)

No additional comments.: # (Please put the score in the following line, e.g. `Score = 16.5`)
Score = 18

## Final project feedback
|                                                                                    | Quality level | Detailed feedback |
| ---------------------------------------------------------------------------------- | ------------- | ----------------- |
| Title & abstract                                                                   |   Excellent     |           |
| Background                                                                         |  Proficient    |            |
| Research question & hypothesis                                                     |   Proficient     |           |
| Data description                                                                   |  Proficient   |             |
| Data wrangling                                                                     |  Excellent     |          |
| Data visualizations                                                                |  Proficient    |             |
| Data analysis and results                                                          |  Excellent     |          |
| Ethics                                                                             | Proficient    |           |
| Conclusion and discussion                                                          |  Proficient     |         |
| Documentation and written communication                                            | Proficient    |           |
| Video                       |      Proficient |            |
| Quality modifier (can result in extra credit beyond 18 points for this assignment) |        |     0        |

### Comments

No additional comments.

## Rubric

Out of 18 points in total.  15 points in the report.  3 points in the video summary of your report for a non-technical audience. There exists the possibility for up to 0.5 points extra credit resulting in a max score of 18.5/18

For the report:
- Each Developing  => -0.5 pts
- Each Unsatisfactory/Missing => -1 pts
  - until the score is 0

For the video:
- Developing  => -1.5 pts
- Unsatisfactory/Missing => -3 pts

|                                                                                    | Unsatisfactory                                                                                                                                                                                                                                                                                      | Developing                                                                                                                                                                                                                                                                                                                          | Proficient                                                                                                                                                                                                                                                    | Exellent                                                                                                                                                                                                                                                                                                                                                          |
| ---------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Title & abstract                                                                   | Title or abstract is confusing or fails to offer important details about the question, variables, context, or methods of the project.                                                                                                                                                               | Title or abstract lacks relevance or fails to offer appropriate details about the question, variables, context, or methods of the project.                                                                                                                                                                                          | Title and abstract are relevant, offering sufficient details about the research project.                                                                                                                                                                      | Title and abstract are informative, succinct, and clear. It offers specific details about the question, variables, context, and proposed methods of the study.                                                                                                                                                                                                    |
| Background                                                                         | Did not have at least 3 reliable and relevant sources. Or relevant sources were not used in relevant ways                                                                                                                                                                                           | A key component was not connected to the research literature. Selected literature was from unreliable sources. Literary supports were vague or ambiguous.                                                                                                                                                                           | Key research components were connected to relevant, reliable literature.                                                                                                                                                                                      | Narrative integrates critical and logical details from the peer-reviewed or other professional research literature. Each key research component is grounded to the literature. Attention is given to different perspectives, threats to validity, and opinion vs. evidence.                                                                                       |
| Research question & hypothesis                                                     | Research issue remains unclear. The research purpose, questions, hypotheses, definitions or variables and controls are still largely undefined, or when they are are poorly formed, ambiguous, or not logically connected to the description of the problem. Unclear connections to the literature. | Research issue is identified, but statement is too broad or fails to establish the importance of the problem. The research purpose, questions, hypotheses, definitions or variables and controls are poorly formed, ambiguous, or not logically connected to the description of the problem. Unclear connections to the literature. | Identifies a relevant research issue. Research questions are succinctly stated, connected to the research issue, and supported by the literature. Variables and controls have been identified and described. Connections are established with the literature. | Presents a significant research problem related to the chemical sciences. Articulates clear, reasonable research questions given the purpose, design, and methods of the project. All variables and controls have been appropriately defined. Proposals are clearly supported from the research and theoretical literature. All elements are mutually supportive. |
| Data description                                                                   | Did not have references to relevant data sources for this problem. Did not describe the data obtained at those sources                                                                                                                                                                              | A key data source was not referenced or described in satisfactory level of detail                                                                                                                                                                                                                                                   | All relevant data sources were referenced and described in terms of their key variables and size                                                                                                                                                              | Multiple data sources for each aspect of the project, All data sources are fully described and referenced. The details of the descriptions also make it clear how they support the needs ot the project.                                                                                                                                                          |
| Data wrangling                                                                     | Data obviously not tidy and not clean. Or even if they are clean/tidy, there was no demonstration of that fact                                                                                                                                                                                      | Data has some minor problems with tidiness or cleanliness. Potentially these are not problems but it wasn't fully demonstrated to be the case in the code                                                                                                                                                                           | Data are made tidy and clean, or are clearly demonstrated to be so from download                                                                                                                                                                              | Data wrangling is complete, but also particularly thoroughly done for a very tough case. Often free text parsing is a bear like this, and doing it well can mean a million edge cases.                                                                                                                                                                            |
| Data visualizations                                                                | Many plots don't capture useful information, or their display of information is difficult to understand.. Legends, labels, axes, titles, captions, etc of many plots may be poor or missing. Figures are not referenced in the main text.                                                           | Most plots make sense and capture useful information. Legends, labels, axes, titles, captions, etc are mostly legible and adequately describe the content of the figure. Most figures are referenced or explained inside the main text and not just left hanging.                                                                   | Plots make sense and capture useful information. Legends, labels, axes, titles, captions, etc are clearly legible and adequately describe the content of the figure. Figures are referenced or explained inside the main text and not just left hanging.      | Plots are beautiful and information dense. Legends, labels, axes, titles, captions, etc clearly describe the content of the figure. Figures are well integrated with the text.                                                                                                                                                                                    |
| Data analysis and results                                                          | Lacks any metrics or at least metrics that are relevant to the problem                                                                                                                                                                                                                              | Some metrics selected are not appropriate for the problem or the metrics selected are not described sufficiently                                                                                                                                                                                                                    | All metrics described are appropriate and and described sufficiently                                                                                                                                                                                          | The metrics are described clearly and succinctly. Their appropriateness for addressing the research problem is clearly described.                                                                                                                                                                                                                                 |
| Ethics                                                                             | Only tries a baseline model, never advances to proper model/hyper-parameter selection                                                                                                                                                                                                               | Only investigates one minimal aspect of their problem, e.g. just did a single hyperparameter selection on a single model                                                                                                                                                                                                            | Does a good model/hyper-parameter selection. Only attempts minimal secondary analysis                                                                                                                                                                         | Does a good model/hyper-parameter selection. Performs multiple secondary analysis such as learning curves, heat maps looking at where in the parameter space things are good/bad, uses statistical testing                                                                                                                                                        |
| Conclusion and discussion                                                          | Doesn't have a narrative/conclusions of what the report accomplished. Or very minimal conclusions, like "X was best" without anything else                                                                                                                                                          | Has narrative/conclusions, but it feels perfunctory and lacks context. Especially if the limitations and ethics sections are largely junk or missing                                                                                                                                                                                | Properly recapitulates the results and provides context for them. Has a sense of the limitatins and ethical issues.                                                                                                                                           | Clearly recapitulates the results and provides context, including the literature/background Has a sense of what to do next, and has good explorations of the limitations and ethics.                                                                                                                                                                              |
| Documentation and written communication                                            | Code documentation is non-existent or perfunctory. Writing has usage errors, poor fluency, and poor rhetoric. If doc/writing are both like this you get unsatisfacory. If one is unsatisfactory and the other is developing you still get unsatisfactory overall.                                   | Code documentation is only partially there. Writing has some usage errors, some poor fluency, or some poor rhetoric. If doc/writing are both like this you get developing. If one is developing and the other is satisfactory you still get developing.                                                                             | Code is documented throughout. Writing has good fluency, good rhetoric, and at most a few minor errors.                                                                                                                                                       | Code documentation is highly informative with docstrings. Writing is high quality. If both doc/writing are like this you get excellent. If one is satisfactory and the other excellent you still get satisfactory.                                                                                                                                                |
| Video (developing -1.5 points, unsatisfactory -3 points)                           | Video well outside the 3-5min time range. Does not communicate major results at a non-technical level. Has major technical issues such as no sound or not available at link. Unsatisfactory if 2/3 of these are true                                                                                | Video just outside the 3-5min time range. Does not communicate well on some points (either wrong tech level or just confusing). Has technical issues such as poor speech quality on video. Developing if 2/3 of these are true                                                                                                      | Video has good quality and is in within 3-5 min range. Communicates the major results at an accessible level for non-technical people.                                                                                                                        | Video has high production quality and is in within 3-5 min range. Communicates the major results clearly and gives a sense of the methods at an accessible level for non-technical people. Has a wow factor (e.g. funny or original approach)                                                                                                                     |
| Quality modifier (can result in extra credit beyond 18 points for this assignment) | Almost everything was just barely proficient. Lots of minimal effort work. Do not double punish; if they already lost points on the rubric do not invoke this: -0.5 points                                                                                                                        |                                                                                                                                                                                                                                                                                                                                     |                                                                                                                                                                      |                                                                                                                                                                                                                                                                                                                       More than 5 excellents on this rubric: +0.5 points                                                                                   |

## Comments
